lzma-rust2/src/enc/lzma2_writer.rs

```diff
@@ -135,7 +135,7 @@ const COMPRESSED_SIZE_MAX: u32 = 64 << 10;
 
 pub fn get_extra_size_before(dict_size: u32) -> u32 {
     if COMPRESSED_SIZE_MAX > dict_size {
```

Comment:
The whole if block is `saturating_sub` in disguise!

So the function should look like this:

```rust
pub fn get_extra_size_before(dict_size: u32) -> u32 {
    COMPRESSED_SIZE_MAX.saturating_sub(dict_size)
}
```